src/converter/one/convert/TimeIntervals.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package one.convert;
+
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.TreeMap;
+import java.util.Arrays;
+
+public final class TimeIntervals {
+    // No overlapping intervals
+    private final TreeMap<Long, Long> timeIntervals = new TreeMap<>();
+
+    public void add(long startInstant, long endInstant) {
+        if (startInstant > endInstant) {
+            throw new IllegalArgumentException("'startInstant' should not be after 'endInstant'");
+        }
+
+        // Are there shorter intervals which overlap with the new interval?
+        NavigableMap<Long, Long> view = timeIntervals.subMap(startInstant, true /* inclusive */, endInstant, true /* inclusive */);
+        Map.Entry<Long, Long> last = view.pollLastEntry();
+        if (last != null) {
+            endInstant = Long.max(last.getValue(), endInstant);
+        }
+        view.clear();
+
+        // Perhaps the end of the interval before 'view' ends after startInstant?
+        Map.Entry<Long, Long> floor = timeIntervals.floorEntry(startInstant);
+        if (floor != null) {
+            long floorEnd = floor.getValue();
+            if (floorEnd >= startInstant) {
+                timeIntervals.remove(floor.getKey());
+                startInstant = floor.getKey();
+                endInstant = Long.max(endInstant, floorEnd);
+            }
+        }
+
+        timeIntervals.put(startInstant, endInstant);
+    }
+
+    public TimeIntervalsList asList() {
+        long[] startIntervals = new long[timeIntervals.size()];
+        long[] endIntervals = new long[timeIntervals.size()];
+        int index = 0;
+        for (Map.Entry<Long, Long> entry : timeIntervals.entrySet()) {
+            startIntervals[index] = entry.getKey();
+            endIntervals[index] = entry.getValue();
+            ++index;
+        }
+        return new TimeIntervalsList(startIntervals, endIntervals);
+    }
+
+    public static final class TimeIntervalsList {
```

Comment:
Exposing a nested class with a repetitive name `TimeIntervals.TimeIntervalsList` looks weird.

A better idea would be to expose `TimeIntervals` for lookup and have `TimeIntervals.Builder` for construction. But I would just use the same `TimeIntervals` class for both purposes despite it's not how public API is usually designed. It's not a public API though.
Anyway, I'm OK with both approaches, as long as it's not `TimeIntervals.TimeIntervalsList`.